File: pe-gui/src/main/java/org/saar/gui/graphics/Graphics.java
package org.saar.gui.graphics;

import org.saar.gui.style.colour.ReadonlyColour;
import org.saar.maths.objects.Polygon;

public interface Graphics {

    void setColour(ReadonlyColour colour);

    void drawLine(int x1, int y1, int x2, int y2);

    void drawRectangle(int x, int y, int w, int h);

    void fillRectangle(int x, int y, int w, int h);

    void drawOval(int cx, int cy, int a, int b);

    void fillOval(int cx, int cy, int a, int b);

    void fillPolygon(Polygon polygon);

    void clear(ReadonlyColour clearColour);

    void process();

    void delete();

}

File: pe-gui/src/main/kotlin/org/saar/gui/graphics/Graphics.kt
package org.saar.gui.graphics

import org.saar.gui.style.Color
import org.saar.maths.objects.Polygon

interface Graphics {

    var color: Color

    fun drawLine(x1: Int, y1: Int, x2: Int, y2: Int)

    fun drawRectangle(x: Int, y: Int, w: Int, h: Int)

    fun fillRectangle(x: Int, y: Int, w: Int, h: Int)

    fun drawOval(cx: Int, cy: Int, a: Int, b: Int)

    fun fillOval(cx: Int, cy: Int, a: Int, b: Int)

    fun fillPolygon(polygon: Polygon)

    fun clear(clearColor: Color)

    fun process()

    fun delete()
}